### DISCUSSION TITLE
Authentication using WWW-Authenticate

### DISCUSSION THREAD

**Discussion by pfeatherstone** [uNetworking/uWebSockets#1821] [Q&A]
Is there a way to do authentication using https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication 
So server initially responds with a 401 Response and a WWW-Authenticate header, waits for a request which includes an Authorization header and user login, check if it matches, then carry on. Is this built in somehow in the framework? Or is there a way to do it with the existing API ?

**Comment by uNetworkingAB** [comment]
uWS is just a server, yes it certainly allows implementing this behavior, definitely, but it's not included in this project.

**Reply by pfeatherstone** [reply]
Could you provide some guidance on how this could be achieved using the existing API ?

**Reply by pfeatherstone** [reply]
Actually I think it's pretty easy to use. Cheers
